install.sh

```diff
@@ -22,86 +84,342 @@ deps=(
-        echo
-      else
-        sudo touch $FILE
+  if [[ $? > 0 ]]
+    then
+      echo "make error check logs"
+    exit
+  else
+    echo "make ok"
+    echo
+  fi
+
+  #run make install
+  echo "beginning make install"
+  sudo make install
+
+  if [[ $? > 0 ]]
+    then
+      echo "install error check logs"
+    exit
+  else
+    echo "installed ok"
+    echo
+  fi
+
+  #run ldconfig
+  sudo ldconfig
+  cd ../../dash
+
+else
+	echo "skipping gstreamer"
+fi
+
+
+
+###############################  openauto  #########################
+if [ "$openauto" = true ]; then
+  echo "installing openauto"
+  cd ..
 
-        # OPEN USB RULE, CREATE MORE SECURE RULE IF REQUIRED
-        echo "SUBSYSTEM==\"usb\", ATTR{idVendor}==\"*\", ATTR{idProduct}==\"*\", MODE=\"0660\", GROUP=\"plugdev\"" | sudo tee $FILE
+  clone openauto
+  git clone $openautoRepo
+  if [[ $? > 0 ]]
+    then
+      cd openauto
       if [[ $? > 0 ]]
         then
-          echo "unable to create permissions"
-        else
-          echo "permissions created"
+          echo "clone/pull error"
+        exit
+      else
+        git pull $openautoRepo
+        echo "cloned OK"
+        cd ..
+        echo
       fi
-    fi
+  else
+    echo "cloned OK"
+    echo
+  fi
+
+  cd openauto
+
+  #beginning cmake
+  cmake -DRPI_BUILD=true -DGST_BUILD=true
+  if [[ $? > 0 ]]
+    then
+      echo "cmake error check logs"
+    exit
+  else
+    echo "cmake OK"
+    echo
+  fi
+
+  #beginning cmake
+  make
+  if [[ $? > 0 ]]
+    then
+      echo "make error check logs"
+    exit
+  else
+    echo "make OK"
+    echo
+  fi
+
+  #run make install
+  echo "beginning make install"
+  sudo make install
+  if [[ $? > 0 ]]
+    then
+      echo "install error check logs"
+    exit
+  else
+    echo "installed ok"
+    echo
+  fi
+  cd ../dash
+else
+	echo "skipping openauto"
 fi
 
-#Start app
-echo "starting app"
-cd ../bin
-./ia
+
+###############################  dash  #########################
+if [ "$dash" = true ]; then
+  #loop through dependencies and install
+	echo "installing dash"
+  echo "making ilclient"
+  make /opt/vc/src/hello_pi/libs/ilclient
+  if [[ $? > 0 ]]
+    then
+      echo "error making ilclient check logs"
+    exit
+  else
+    echo "make ok"
+    echo
+  fi
+
+  echo "running cmake for dash"
+  cmake -DRPI_BUILD=TRUE -DGST_BUILD=TRUE .
+  if [[ $? > 0 ]]
+    then
+      echo "cmake error check logs"
+    exit
+  else
+    echo "cmake OK"
+    echo
+  fi
+
+  echo "running make"
+  make
+  if [[ $? > 0 ]]
+    then
+      echo "make error check output above"
+      exit
+    else
+      echo "make ok, executable can be found ../bin/ia"
+      echo
+
+      #check if usb rules exist
+      echo "checking if permissions exist"
+
+      #udev rule to be created below, change as needed
+      FILE=/etc/udev/rules.d/51-iadash.rules
+      if [[ -f "$FILE" ]]
+        then
+          echo "rules exists"
+          echo
+        else
+          sudo touch $FILE
```

Comment:
This is unneeded. `tee` will create the file and if you do it this way and the `echo` or `tee` below fails, a subsequent run will errantly detect that the rules exist and produce a broken product.

Reply:
updated

https://github.com/openDsh/dash/blob/a70d382506d021b17a2d9e41465acccda00c192d/install.sh#L367-L377